runtime/stdlib/assert.go

```diff
@@ -0,0 +1,71 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright 2019-2022 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package stdlib
+
+import (
+	"github.com/onflow/cadence/runtime/interpreter"
+	"github.com/onflow/cadence/runtime/sema"
+)
+
+// AssertFunction
+
+const assertFunctionDocString = `
+Terminates the program if the given condition is false, and reports a message which explains how the condition is false.
+Use this function for internal sanity checks.
+
+The message argument is optional.
+`
+
+var assertFunctionType = &sema.FunctionType{
+	Parameters: []*sema.Parameter{
+		{
+			Label:          sema.ArgumentLabelNotRequired,
+			Identifier:     "condition",
+			TypeAnnotation: sema.NewTypeAnnotation(sema.BoolType),
+		},
+		{
+			Identifier:     "message",
+			TypeAnnotation: sema.NewTypeAnnotation(sema.StringType),
+		},
+	},
+	ReturnTypeAnnotation: sema.NewTypeAnnotation(
+		sema.VoidType,
+	),
+	RequiredArgumentCount: sema.RequiredArgumentCount(1),
+}
+
+var AssertFunction = NewStandardLibraryFunction(
+	"assert",
+	assertFunctionType,
+	assertFunctionDocString,
+	func(invocation interpreter.Invocation) interpreter.Value {
+		result := invocation.Arguments[0].(interpreter.BoolValue)
```

Comment:
Check this cast

Reply:
This is just moving code, but I added them separately in #1422. Will also add them here I guess

Reply:
Added in 8790562c756b4bf654d20b6da95cfe5ec8dda56c